pkg/nodes/enricher/enricher_impl.go

```diff
@@ -49,7 +49,22 @@ func (e *enricherImpl) RemoveNodeIntegration(id string) {
 
 // EnrichNode enriches a node with the integration set present.
 func (e *enricherImpl) EnrichNode(node *storage.Node) error {
+	nodeNoteSet := make(map[storage.Node_Note]struct{}, len(node.GetNotes()))
+	for _, note := range node.GetNotes() {
+		nodeNoteSet[note] = struct{}{}
+	}
```

Comment:
I am thinking (part 1) you could filter out `storage.Node_MISSING_SCAN_DATA` from `node.Notes`, leaving all notes except `storage.Node_MISSING_SCAN_DATA`.

Reply:
I copied this over from how we handle image notes. I don't really recall the reason for doing it this way. I will need to see if I can find if there was a compelling reason I'm forgetting. If not, then I'll take your suggestion

Reply:
I think it was originally meant to ensure we do not delete notes which are not added in this function, but every note is added here, so I will switch to your suggestion here, thanks